connectors/src/connectors/intercom/lib/conversation_permissions.ts

```diff
@@ -90,55 +93,81 @@ export async function retrieveIntercomConversationsPermissions({
+    permission: "none",
+    dustDocumentId: null,
+    lastUpdatedAt: null,
+  };
+
+  const teamswithReadPermission = await IntercomTeam.findAll({
+    where: {
+      connectorId: connectorId,
+      permission: "read",
+    },
+  });
 
-  // If Root level we retrieve the list of Help Centers.
+  // If Root level we display the fake parent "Conversations"
   // If isReadPermissionsOnly = true, we retrieve the list of Teams from DB that have permission = "read"
   // If isReadPermissionsOnly = false, we retrieve the list of Teams from Intercom
-  if (isRootLevel) {
-    const teamsFromDb = await IntercomTeam.findAll({
-      where: {
-        connectorId: connectorId,
-        permission: "read",
-      },
-    });
-
-    if (isReadPermissionsOnly) {
-      resources = teamsFromDb.map((team) => ({
-        provider: connector.type,
-        internalId: getTeamInternalId(connectorId, team.teamId),
-        parentInternalId: null,
-        type: "channel",
-        title: team.name,
-        sourceUrl: null,
-        expandable: false,
-        permission: team.permission,
-        dustDocumentId: null,
-        lastUpdatedAt: null,
-      }));
-    } else {
+  if (isReadPermissionsOnly) {
+    if (isRootLevel && teamswithReadPermission.length > 0) {
+      resources.push(rootConversationRessource);
+    }
+    if (parentInternalId === teamsInternalId) {
+      teamswithReadPermission.forEach((team) => {
+        resources.push({
+          provider: connector.type,
+          internalId: getTeamInternalId(connectorId, team.teamId),
+          parentInternalId: teamsInternalId,
+          type: "folder",
+          title: team.name,
+          sourceUrl: null,
+          expandable: false,
+          permission: team.permission,
+          dustDocumentId: null,
+          lastUpdatedAt: null,
+        });
+      });
+    }
+  } else {
+    if (isRootLevel) {
+      resources.push(rootConversationRessource);
+    }
+    if (parentInternalId === teamsInternalId) {
       const teams = await fetchIntercomTeams(intercomClient);
-      resources = teams.map((team) => {
-        const isTeamInDb = teamsFromDb.some((teamFromDb) => {
+      teams.forEach((team) => {
+        const isTeamInDb = teamswithReadPermission.some((teamFromDb) => {
           return teamFromDb.teamId === team.id;
         });
-        return {
+        resources.push({
           provider: connector.type,
           internalId: getTeamInternalId(connectorId, team.id),
-          parentInternalId: null,
-          type: "channel",
+          parentInternalId: teamsInternalId,
+          type: "folder",
           title: team.name,
           sourceUrl: null,
           expandable: false,
           permission: isTeamInDb ? "read" : "none",
           dustDocumentId: null,
           lastUpdatedAt: null,
-        };
+        });
       });
     }
-    resources.sort((a, b) => {
-      return a.title.localeCompare(b.title);
-    });
   }
+  resources.sort((a, b) => {
```

Comment:
This should definitely be the default behavior and be abstracted in the Connector 🙃.